Assets/MixedRealityToolkit/_Core/Utilities/UX/HoverLight.cs

```diff
@@ -92,21 +100,87 @@ private void OnDrawGizmosSelected()
             Gizmos.DrawIcon(transform.position + Vector3.back * Radius, string.Empty, false);
         }
 
-        private void Initialize()
+        private static void AddHoverLight(HoverLight light)
+        {
+            if (activeHoverLights.Count >= hoverLightCount)
+            {
+                Debug.LogWarningFormat("Max hover light count ({0}) exceeded.", hoverLightCount);
+            }
+
+            activeHoverLights.Add(light);
+        }
+
+        private static void RemoveHoverLight(HoverLight light)
+        {
+            activeHoverLights.Remove(light);
+        }
+
+        private static void Initialize()
+        {
+            _HoverLightDataID = Shader.PropertyToID("_HoverLightData");
+        }
+
+        private static void UpdateHoverLights(bool forceUpdate = false)
         {
-            hoverPositionId = Shader.PropertyToID("_HoverPosition");
-            hoverRadiusId = Shader.PropertyToID("_HoverRadius");
-            hoverColorId = Shader.PropertyToID("_HoverColor");
+            if (lastHoverLightUpdate == -1)
+            {
+                Initialize();
+            }
+
+            if (!forceUpdate && (Time.frameCount == lastHoverLightUpdate))
+            {
+                return;
+            }
+
+            if (activeHoverLights.Count > 1)
+            {
+                Shader.EnableKeyword("_MULTI_HOVER_LIGHT");
```

Comment:
Let's make this a const string seeing as we use it in multiple places.

We could do the same of for all the keywords.

Reply:
if we are using it externally, let's make it a static readonly string so that the reference is fetched and not the literal constant.

Reply:
This string is only used by this file at the moment!